AEPLifecycle/Tests/LifecycleDataStoreCacheTests.swift

```diff
@@ -0,0 +1,183 @@
+        lifecycleDataStoreCache = LifecycleDataStoreCache(dataStore: dataStore)
+        
+        let expectedTS = currTimeStamp + TimeInterval(LifecycleConstants.CACHE_TIMEOUT_SECONDS)
+        
+        //test
+        lifecycleDataStoreCache.setLastKnownTimeStamp(ts: currTimeStamp + TimeInterval(1))
+        XCTAssertEqual(expectedTS, lifecycleDataStoreCache.getCloseTimestampSec())
+        // should not update persistence
+        XCTAssertEqual(currTimeStamp, getAppCloseTSFromPersitence())
+    }
+    
+    func testSettLastKnownTS_DifferenceMoreThanCacheTimeoutSinceLastUpdate_WillUpdateValueInPersitenceAndReflectInNextLaunch() {
+        //setup
+        let currTimeStamp = NSDate().timeIntervalSince1970
+        persistAppCloseTS(ts: currTimeStamp)
+        lifecycleDataStoreCache = LifecycleDataStoreCache(dataStore: dataStore)
+        
+        let expectedTS = currTimeStamp + TimeInterval(LifecycleConstants.CACHE_TIMEOUT_SECONDS)
+        let expectedTS2 = currTimeStamp + TimeInterval(3) + TimeInterval(LifecycleConstants.CACHE_TIMEOUT_SECONDS)
+        XCTAssertEqual(expectedTS, lifecycleDataStoreCache.getCloseTimestampSec())
+        
+        //test
+        lifecycleDataStoreCache.setLastKnownTimeStamp(ts: currTimeStamp + TimeInterval(3))
+        
+        // verify that datastore is updated
+        XCTAssertEqual(currTimeStamp + TimeInterval(3), getAppCloseTSFromPersitence())
+        
+        lifecycleDataStoreCache = LifecycleDataStoreCache(dataStore: dataStore)
+        // verify the closeTimeStamp value in next launch
+        XCTAssertEqual(expectedTS2, lifecycleDataStoreCache.getCloseTimestampSec())
+    }
+    
+    func testSettLastKnownTS_ConsecutiveUpdates() {
+        //setup
+        let currTimeStamp = NSDate().timeIntervalSince1970
+        persistAppCloseTS(ts: currTimeStamp)
+        lifecycleDataStoreCache = LifecycleDataStoreCache(dataStore: dataStore)
+        
+        let expectedTS = currTimeStamp + TimeInterval(LifecycleConstants.CACHE_TIMEOUT_SECONDS)
+        XCTAssertEqual(expectedTS, lifecycleDataStoreCache.getCloseTimestampSec())
+        
+        //test
+        lifecycleDataStoreCache.setLastKnownTimeStamp(ts: currTimeStamp + TimeInterval(1))
+        lifecycleDataStoreCache.setLastKnownTimeStamp(ts: currTimeStamp + TimeInterval(2))
+        lifecycleDataStoreCache.setLastKnownTimeStamp(ts: currTimeStamp + TimeInterval(3))
+        lifecycleDataStoreCache.setLastKnownTimeStamp(ts: currTimeStamp + TimeInterval(4))
+        lifecycleDataStoreCache.setLastKnownTimeStamp(ts: currTimeStamp + TimeInterval(5))
+        lifecycleDataStoreCache.setLastKnownTimeStamp(ts: currTimeStamp + TimeInterval(6))
+        
+        // verify that datastore is updated
+        XCTAssertEqual(currTimeStamp + TimeInterval(6), getAppCloseTSFromPersitence())
+    }
+    
+    func testSetAppStartTS() {
+        //setup
+        let currTimeStamp = NSDate().timeIntervalSince1970
```

Comment:
can we move `currentTimestamp` this in the setup step and reuse it in all tests?